src/sphinx_book_theme/_transforms.py

```diff
@@ -0,0 +1,70 @@
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from typing import Any
+from docutils import nodes as docutil_nodes
+from sphinx import addnodes as sphinx_nodes
+from .nodes import SideNoteNode
+import copy
+
+
+class HandleFootnoteTransform(SphinxPostTransform):
+
+    default_priority = 1
+    formats = ("html",)
+
+    def run(self, **kwargs: Any) -> None:
+        theme_options = self.env.config.html_theme_options
+        if theme_options.get("use_sidenotes", False) is False:
+            return None
+        for node in self.document.traverse(docutil_nodes.footnote_reference):
+            parent = None
+            for ftnode in self.document.traverse(docutil_nodes.footnote):
+                # matching the footnote reference with footnote
+                if (
+                    len(ftnode.attributes["backrefs"])
+                    and ftnode.attributes["backrefs"][0] == node.attributes["ids"][0]
+                ):
+                    parent = ftnode.parent
+                    text = ftnode.children[1].astext()
```

Comment:
Explain why we do `children[1]`. What is the first element of `children`? What is the element we want to grab? This can be helpful in case docutils changes this ordering in the future

Reply:
done in [f5efddf](https://github.com/executablebooks/sphinx-book-theme/pull/546/commits/f5efddf4772fa3d2f39b3755ecabe5d228b90e54)